spark/src/main/java/io/delta/dynamodbcommitstore/integration_tests/dynamodb_commitowner_integration_test.py

```diff
@@ -0,0 +1,236 @@
+export AWS_DEFAULT_REGION=us-west-2
+
+# ===== Optional input from user =====
+export DELTA_CONCURRENT_WRITERS=20
+export DELTA_CONCURRENT_READERS=2
+export DELTA_NUM_ROWS=200
+export DELTA_DYNAMO_ENDPOINT=https://dynamodb.us-west-2.amazonaws.com
+
+# ===== Optional input from user (we calculate defaults using S3_BUCKET and RUN_ID) =====
+export RELATIVE_DELTA_TABLE_PATH=___
+export DELTA_DYNAMO_TABLE_NAME=___
+
+./run-integration-tests.py --use-local --run-dynamodb-commit-owner-integration-tests \
+    --dbb-conf io.delta.storage.credentials.provider=com.amazonaws.auth.profile.ProfileCredentialsProvider \
+               spark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.profile.ProfileCredentialsProvider
+"""
+
+# ===== Mandatory input from user =====
+run_id = os.environ.get("RUN_ID")
+s3_bucket = os.environ.get("S3_BUCKET")
+
+# ===== Optional input from user =====
+concurrent_writers = int(os.environ.get("DELTA_CONCURRENT_WRITERS", 2))
+concurrent_readers = int(os.environ.get("DELTA_CONCURRENT_READERS", 2))
+num_rows = int(os.environ.get("DELTA_NUM_ROWS", 16))
+dynamo_endpoint = os.environ.get("DELTA_DYNAMO_ENDPOINT", "https://dynamodb.us-west-2.amazonaws.com")
+
+# ===== Optional input from user (we calculate defaults using RUN_ID) =====
+relative_delta_table_path = os.environ.get("RELATIVE_DELTA_TABLE_PATH", f"tables/table_ddb_cs_{run_id}_{str(uuid.uuid4())}")\
+    .rstrip("/")
+dynamo_table_name = os.environ.get("DELTA_DYNAMO_TABLE_NAME", "test_ddb_cs_table_" + run_id)
+
+delta_table_path = "s3a://" + s3_bucket + "/" + relative_delta_table_path
+relative_delta_log_path = relative_delta_table_path + "/_delta_log/"
+
+if delta_table_path is None:
+    print(f"\nSkipping Python test {os.path.basename(__file__)} due to the missing env variable "
+          f"`DELTA_TABLE_PATH`\n=====================")
+    sys.exit(0)
+
+dynamodb_commit_owner_conf = json.dumps({
+    "managedCommitsTableName": dynamo_table_name,
+    "dynamoDBEndpoint": dynamo_endpoint
+})
+
+test_log = f"""
+==========================================
+run id: {run_id}
+delta table path: {delta_table_path}
+dynamo table name: {dynamo_table_name}
+
+concurrent writers: {concurrent_writers}
+concurrent readers: {concurrent_readers}
+number of rows: {num_rows}
+
+relative_delta_table_path: {relative_delta_table_path}
+relative_delta_log_path: {relative_delta_log_path}
+==========================================
+"""
+print(test_log)
+
+commit_owner_property_key = "managedCommits.commitOwner"
+property_key_suffix = "-preview"
+
+spark = SparkSession \
+    .builder \
+    .appName("utilities") \
+    .master("local[*]") \
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension") \
+    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog") \
+    .config(f"spark.databricks.delta.properties.defaults.{commit_owner_property_key}{property_key_suffix}", "dynamodb") \
```

Comment:
Could we also run the same test on top of a new table which is not a managed-commit table at the time of creation?
So CREATE -> INSERT+SELECT -> UPGRADE -> INSERT+SELECT -> DOWNGRADE. -> INSERT+SELECT -> UPGRADE

Reply:
Done